### DISCUSSION TITLE
Sugerencias para optimizar el entrenamiento

### DISCUSSION THREAD

**Discussion by JavierBZ** [JavierBZ/Tarea3-IIC3697#24] [Announcements]
Hola, ya varios me han preguntado sobre como mejorar el rendimiento en cuanto a la cantidad de iteraciones por segundo al comenzar el entrenamiento de la red. Les puedo dar las siguientes sugerencias:

1. Optimicen la función sample del replay memory. Para ello les recomiendo usar la función random.sample, la cual permite obtener el batch de experiencias en a lo más dos lineas.
2. Para transformar a tensores utilicen torch.fom_numpy() en vez de torch.tensor(), es más eficiente.
3. Revisen bien y optimicen todas las operaciones de tensores que hagan, tanto en la red Dueling DQN como en el cálculo de la loss.
4. No olviden activar el uso de GPU en Colab y hacer correctamente el paso de los tensores a 'cuda' con `.to(device)`.
5. Por último, no esta de más recordarles que no utilicen Fors!!!!, cualquier uso de for ya sea en su replay memory o en operaciones de tensores les puede enlentecer mucho el rendimiento.

Espero les sirva y si alguien tiene alguna recomendación extra por favor coméntela.
